src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -77,6 +80,10 @@
  */
 public abstract class Text implements TextRepresentable {
 
+    static {
+        TypeSerializers.getDefaultSerializers().registerType(TypeToken.of(Text.class), new TextConfigSerializer());
```

Comment:
Totally forgot about this :p I'll look at it later although I have pretty much zero prior experience with the data API.